packages/synapse-react-client/src/utils/functions/TableColumnSchemaUtils.ts

```diff
@@ -4,40 +4,46 @@ import {
   TableUpdateTransactionRequest,
 } from '@sage-bionetworks/synapse-types'
 import { SetOptional } from 'type-fest'
-import { isEqual } from 'lodash-es'
 import { createColumnModels } from '../../synapse-client'
+import { isEqualTreatUndefinedAsMissing } from './IsEqualTreatUndefinedAsMissing'
 
 /**
- * Creates the column models for a change request and returns
- * the change request to be passed to the Synapse backend.
- * @param accessToken
- * @param tableId
- * @param oldSchema
- * @param proposedSchema
+ * Creates the column models for a change request and returns the change request to be passed to the Synapse backend.
+ *
+ * Note that to update a column model and preserve table data, the ID of the
+ * @param accessToken the Synapse access token
+ * @param tableId the ID of the table to change
+ * @param oldSchema the current schema of the table
+ * @param proposedSchema the proposed schema of the table. For TableEntities, the IDs should be preserved for columns
+ *                       that should maintain their data, even if the column model has changed
  */
 export async function createTableUpdateTransactionRequest(
   accessToken: string,
   tableId: string,
   oldSchema: ColumnModel[],
   proposedSchema: SetOptional<ColumnModel, 'id'>[],
 ): Promise<TableUpdateTransactionRequest> {
-  // Create any models that do not have an ID, or that have changed
+  // Keep track of the old column models by ID
   const oldColumnModelId2Model: Map<string, ColumnModel> = new Map()
   for (const columnModel of oldSchema) {
     oldColumnModelId2Model.set(columnModel.id, columnModel)
   }
 
-  // The newSchema will match the proposed schema, but column models that have been changed will have the ID stripped
+  // Create the new column models.
+  // The newSchema will match the proposed schema, but column models that have been changed will end up getting a new ID
   let newSchema: SetOptional<ColumnModel, 'id'>[] = []
   for (const m of proposedSchema) {
     const copy: SetOptional<ColumnModel, 'id'> = {
       ...m,
     }
     if (copy.id != null) {
-      // any changes to the existing column model?
+      // If there are any changes to the existing column model, remove the ID before we create the new one
       const oldColumnModel: ColumnModel | undefined =
         oldColumnModelId2Model.get(copy.id)
-      if (oldColumnModel != undefined && !isEqual(oldColumnModel, copy)) {
+      if (
+        oldColumnModel != undefined &&
+        !isEqualTreatUndefinedAsMissing(oldColumnModel, copy)
+      ) {
```

Comment:
This was one bug. These columns are equivalent, but our code did not treat them as equal until changing the `isEqual` method

```js
{
  "id": "123",
  "name": "foo",
  "columnType": "STRING"
}
```

```js
{
  "id": "123",
  "name": "foo",
  "columnType": "STRING",
  "maximumLength": undefined
}
```